### ISSUE TITLE
YOLO11 seem not stable as old YOLOV8, at least bbox on large objects

### ISSUE THREAD

**Issue by AI-Passionner** [ultralytics/ultralytics#22457]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

@Y-T-G and @glenn-jocher, Thanks to your team for this great OB system. I have been training many object detection models, very successful. 
However, I like to share one use case where YOLO11 seems not stable, underperforming the old YOLOV8, which is very stable, regardless of whether the objects are big or small.
On the other hand, I am disappointed about the YOLO11, at least on one of the projects, detecting card edge and inner bourdary and both are big objects. Both YOLO11 and old YOLOv8 give similare metric report. But they show way different on the bounding box. The old YOLOV8 gives much accurate and tight bbox. 
I am not sure if it is related toa bug. My guess is that it is related to the anchor box, NMS, or something else. 

![Image](https://github.com/user-attachments/assets/f166554f-95bb-4d4d-b70d-e3cdc95ab88f)

The YOLO11 and YOLOV8 versions in the attached image are from ultralytic 8.3.199, while the old YOLOV8 from ultralytic=8.2.92 ( the best so far)

**Comment by UltralyticsAssistant** [comment]
👋 Hello @AI-Passionner, thank you for your detailed report and for using Ultralytics 🚀! This is an automated response to help move things forward quickly — an Ultralytics engineer will also assist you here shortly.

We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) (MRE) to help us debug it. Given your observation about bbox stability differences between YOLO11 and YOLOv8, an MRE is especially helpful. Please include:
-Exact ultralytics package versions you tested (e.g., 8.2.92 vs 8.3.199) and OS/GPU/CUDA details
-Exact model variants and checkpoints used (e.g., YOLO11n/s/m vs YOLOv8n/s/m)
-Full training/evaluation commands and all overrides (imgsz, epochs, batch, lr, augment, optimizer)
-Inference/NMS settings (conf, iou, max_det, agnostic, classes) used to generate the shown results
-Any anchor-related settings (anchor_free/autoanchor) and task config differences
-A tiny dataset subset or a few representative images with labels where the issue is visible
-Training and validation logs plus sample predictions from both models on the same images

If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/). Screenshots like the one you shared are great—adding side-by-side predictions from YOLO11 and YOLOv8 on the same images with identical conf/iou also helps a lot 🙏

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [Python>=3.8](https://www.python.org/) environment with [PyTorch>=1.8](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

-Notebooks with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
-Google Cloud Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
-Amazon Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
-Docker Image. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

Thanks again for the thorough comparison and screenshot — once you share the MRE details above, we can quickly investigate potential differences related to NMS, anchors, or configuration between YOLO11 and YOLOv8 🔍✨

**Comment by glenn-jocher** [comment]
Thanks for the detailed example—happy to dig in. Note that anchors aren’t involved here, as both YOLOv8 and YOLO11 use anchor-free heads; see the YOLO11 vs YOLOv8 comparison and anchor-free notes in the docs: [YOLO11 vs YOLOv8](https://docs.ultralytics.com/compare/yolo11-vs-yolov8/) and [YOLO11 anchor-free detection](https://www.ultralytics.com/blog/benefits-ultralytics-yolo11-being-anchor-free-detector). To investigate, please share: the full output of `yolo checks`, the exact train/predict commands used for both models, a tiny reproducible sample (3–5 images with labels), both trained weights, and text predictions rather than screenshots; for an apples-to-apples A/B, run the following to save per-box txt/conf with identical thresholds and a larger resolution for big objects, then attach the two `labels/` folders:

```bash
# YOLO11
yolo predict model=path/to/yolo11/best.pt source=path/to/images imgsz=1024 conf=0.25 iou=0.70 save_txt=True save_conf=True

# YOLOv8
yolo predict model=path/to/yolov8/best.pt source=path/to/images imgsz=1024 conf=0.25 iou=0.70 save_txt=True save_conf=True
```

If box tightness is still noticeably different, please also run `yolo val model=... data=... imgsz=1024` for both weights and include the logs so we can compare AP vs size buckets and NMS behavior.
